lib/parlour/conflict_resolver.rb

```diff
@@ -78,12 +78,19 @@ def resolve_conflicts(namespace, &resolver)
                 c.is_a?(RbiGenerator::Include) || c.is_a?(RbiGenerator::Extend)
               end
             end
-            
+
+            # Special case: is there any duplication? If so, remove duplication.
+            if all_eql?(children)
+              Debugging.debug_puts(self, Debugging::Tree.end("All children are identical"))
+
+              namespace.children.uniq! { |m| [m.class, m.name] }
+            end
+
```

Comment:
The check of `all_eql?` is not necessary and it would fail in some cases. Instead, we want to do the deduplication of all the mixin types between their types and with the given name. So two copies of `include I1` and two copies of `extend I1` would collapse to one copy of each.

I will instead propose:
```suggestion
            deduplicate_mixins_of_name(namespace, name)
```
where the implementation of the method is:
```ruby
    sig { params(namespace: RbiGenerator::Namespace, name: String).void }
    # Given a namespace and a child name, removes all duplicate children that are mixins
    # and that have the given name, except the first found instance.
    #
    # @param namespace [RbiGenerator::Namespace] The namespace to deduplicate mixins in.
    # @param name [String] The name of the mixin modules to deduplicate.
    # @return [void]
    def deduplicate_mixins_of_name(namespace, name)
      found_map = {}
      namespace.children.delete_if do |x|
        # ignore children whose names don't match
        next unless x.name == name
        # ignore children that are not mixins
        next unless x.is_a?(RbiGenerator::Include) || x.is_a?(RbiGenerator::Extend)

        delete = found_map.key?(x.class)
        found_map[x.class] = true
        delete
      end
    end
```